cockpit/handlers/lightSource.py

```diff
@@ -147,8 +147,11 @@ def onSaveSettings(self, settings):
     ## Load our settings from the provided dict.
     def onLoadSettings(self, settings):
         if self.name in settings:
-            self.setExposureTime(settings[self.name]['exposureTime'])
-            self.setEnabled(settings[self.name]['isEnabled'])
+            #Only chnbage settings if needed.
+            if (self.getExposureTime != settings[self.name]['exposureTime']):
+                self.setExposureTime(settings[self.name]['exposureTime'])
+            if (self.getIsEnabled != settings[self.name]['isEnabled']):
```

Comment:
parentheses around this condition which is a one line comparison are not needed.